lib/config-generator.js

```diff
@@ -525,6 +526,16 @@ class ConfigGenerator {
         return stats;
     }
 
+    buildWatchOptionsConfig() {
+        const watchOptions = {
+            ignored: /node_modules/
+        };
+
+        this.webpackConfig.watchOptionsConfigurationCallback(watchOptions);
```

Comment:
this should use `applyOptionsCallback` to support the different patterns for the callback (either mutating the argument, or returning a new object), as we do elsewhere.

Reply:
Ah yes you're right, updated.